flang/lib/Semantics/check-omp-structure.cpp

```diff
@@ -2456,6 +2465,19 @@ void OmpStructureChecker::Enter(const parser::OmpClause::Private &x) {
   CheckIntentInPointer(x.v, llvm::omp::Clause::OMPC_private);
 }
 
+void OmpStructureChecker::Enter(const parser::OmpClause::Nowait &x) {
+  CheckAllowed(llvm::omp::Clause::OMPC_nowait);
```

Comment:
This does look for begin or end directive. But for C/C++ we do not have end directive, so nowait clause is present in begin directive list as well in OMP.td. So this will not flag any error for Fortran when nowait is used in begin directive clause list. Removing the nowait clause from begin directive list in OMP.td will flag error for C/C++ tests.

Reply:
Thanks. Since the handling is different due to the C/C++ issue. It will be good to call this out in the patch summary.